### PR TITLE
Refactor experimental features

### PR DESCRIPTION
**Pull request by reuvenperetz**
#### PTQ:
Rename PTQ APIs without 'experimental'.
Remove new_experimental_exporter flag from APIs.
#### GPTQ:
Remove new_experimental_exporter flag from APIs.
Replace GradientPTQConfig with GradientPTQConfigV2 (when V2 is removed from the class name).
#### Tutorials:
Remove mentions of experimental (including in some kpi_data methods fixes that are missing from previous PR).

**Notice**, all tests are now wrapped using MCTQ infrastructure. Thus, some tests had to be fixed as well to adapt to the new output quantized model.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 1]
```diff
@@ -90,13 +90,14 @@ def __init__(self, n_iter: int,
             rounding_type (RoundingType): An enum that defines the rounding type.
             use_hessian_based_weights (bool): Whether to use Hessian-based weights for weighted average loss.
             optimizer_quantization_parameter (Any): Optimizer to override the rest optimizer  for quantizer parameters.
-            optimizer_bias (Any): Optimizer to override the rest optimizer for bias.
+            optimizer_bias (Any): Optimizer to override the rest optimizerfor bias.
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/common/gptq_config.py:93]: Missing space

File: tests/pytorch_tests/layer_tests/base_pytorch_layer_test.py [hunk 2]
```diff
@@ -193,6 +235,9 @@ def compare(self, quantized_model: Module, float_model: Module, input_x=None, qu
 
         # Check inference is possible
         input_tensors = self.generate_inputs()
+        # input_tensors = [t.to('cpu') for t in input_tensors]
+        # quantized_model.to('cpu')
```
Comment (reuvenperetz) [inline tests/pytorch_tests/layer_tests/base_pytorch_layer_test.py:239]: Remove commented out code

File: tests/keras_tests/feature_networks_tests/feature_networks/scale_equalization_test.py [hunk 3]
```diff
@@ -98,8 +98,11 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
         f_first_linear_op_index = 1
         f_second_linear_op_index = 4 + int(self.zero_pad)
 
-        quantized_model_layer1_weight = quantized_model.layers[q_first_linear_op_index].weights[0]
-        quantized_model_layer2_weight = quantized_model.layers[q_second_linear_op_index].weights[0]
+        first_op_attr = 'depthwise_kernel' if isinstance(self.first_op2d, layers.DepthwiseConv2D) else 'kernel'
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/scale_equalization_test.py:101]: use const for depthwise_kernel and kernel?

File: tests/keras_tests/feature_networks_tests/feature_networks/test_kmeans_quantizer.py [hunk 4]
```diff
@@ -125,7 +125,8 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
         # using different methods (but started as the same value)
         conv_layers = get_layers_from_model_by_type(quantized_model, layers.Conv2D)
         self.unit_test.assertTrue(np.sum(
-            np.abs(conv_layers[0].weights[0].numpy() - conv_layers[2].weights[0].numpy())) > 0)
+            np.abs(conv_layers[0].get_quantized_weights()['kernel'] - conv_layers[2].get_quantized_weights()['kernel'])) > 0)
```
Comment (ofirgo) [inline tests/keras_tests/feature_networks_tests/feature_networks/test_kmeans_quantizer.py:128]: same (const)

File: tests/keras_tests/models_tests/test_networks_runner.py [hunk 5]
```diff
@@ -103,25 +103,22 @@ def representative_data_gen():
 
         core_config = mct.core.CoreConfig(quantization_config=qc)
         if self.gptq:
-            arc = mct.gptq.GradientPTQConfig(n_iter=2, optimizer=tf.keras.optimizers.Adam(
+            arc = mct.gptq.GradientPTQConfig(n_epochs=1, optimizer=tf.keras.optimizers.Adam(
```
Comment (ofirgo) [inline tests/keras_tests/models_tests/test_networks_runner.py:106]: maybe set the epochs to 2? I remember having undetected issues in GPTQ that occurred in multiple epochs and we didn't see them because all of our tests were with 1 epoch

File: tests/keras_tests/non_parallel_tests/test_keras_tp_model.py [hunk 6]
```diff
@@ -252,20 +252,16 @@ def test_get_keras_tpc(self):
...
-        quantized_model, _ = mct.ptq.keras_post_training_quantization_experimental(model,
-                                                                                   rep_data,
-                                                                                   core_config=core_config,
-                                                                                   target_kpi=mct.core.KPI(np.inf),
-                                                                                   target_platform_capabilities=tpc,
-                                                                                   new_experimental_exporter=True)
+        quantized_model, _ = mct.ptq.keras_post_training_quantization(model,
+                                                                      rep_data,
+                                                                      target_platform_capabilities=tpc)
+        core_config = mct.core.CoreConfig(mixed_precision_config=mct.core.MixedPrecisionQuantizationConfig(num_of_images=1,
```
Comment (ofirgo) [inline tests/keras_tests/non_parallel_tests/test_keras_tp_model.py:258]: same for number of images (I know that it's not part of the API change, but if you change this and it doesn't break anything that requires debug maybe use this chance and update the test)

File: tests/pytorch_tests/layer_tests/base_pytorch_layer_test.py [hunk 7]
```diff
@@ -202,10 +244,11 @@ def __compare_8bits_quantization_mode(self, float_model, quantized_model, quanti
             op = get_node_operation(node, quantized_model)
             if op == OUTPUT or op == operator.getitem or is_node_fake_quant(node):
                 continue
-            if hasattr(quantized_model, str(node.target)):
+            if has_nested_attr(quantized_model, str(node.target)):
                 if type(op) in PYTORCH_LAYER_TEST_OPS['kernel_ops']:
-                    quantized_weights = get_layer_weights(getattr(quantized_model, node.target))
-                    float_weights = get_layer_weights(getattr(float_model, node.target))
+                    quantized_weights = get_layer_weights(get_nested_attr(quantized_model, node.target))
+                    float_layer_name = str(node.target).split('.')[0]
```
Comment (ofirgo) [inline tests/pytorch_tests/layer_tests/base_pytorch_layer_test.py:250]: add a comment regarding what are you extracting here

File: tests/pytorch_tests/model_tests/feature_models/gptq_test.py [hunk 8]
```diff
@@ -104,8 +104,7 @@ def run_test(self):
             self.representative_data_gen_experimental,
             core_config=self.get_core_config(),
             target_platform_capabilities=self.get_tpc(),
-            gptq_config=self.get_gptq_configv2(),
-            new_experimental_exporter=self.experimental_exporter)
+            gptq_config=self.get_gptq_configv2())
```
Comment (ofirgo) [inline tests/pytorch_tests/model_tests/feature_models/gptq_test.py:107]: still v2?

File: tests/pytorch_tests/model_tests/feature_models/gptq_test.py [hunk 9]
```diff
@@ -201,3 +200,20 @@ def gptq_compare(self, ptq_model, gptq_model, input_x=None):
         # check all weights were not updated in gptq model compared to ptq model
         w_diffs = [np.isclose(np.max(np.abs(w_ptq - w_gptq)), 0) for w_ptq, w_gptq in zip(ptq_weights, gptq_weights)]
         self.unit_test.assertTrue(np.all(w_diffs), msg="GPTQ: some weights were updated in zero learning rate test")
+
+    def _extract_weights(self, model):
```
Comment (ofirgo) [inline tests/pytorch_tests/model_tests/feature_models/gptq_test.py:204]: seems like a useful helper method for tests, maybe put it in some common pytorch tests location so we'll be able to use it in the future?

**Review by ofirgo — APPROVED** [review]
Approved subject to some minor changes
